src/main/java/org/popcraft/popcraft/commands/Listgen.java

```diff
@@ -1,71 +1,64 @@
 package org.popcraft.popcraft.commands;
 
-import org.bukkit.Bukkit;
+import com.google.inject.Inject;
+import io.netty.handler.logging.LogLevel;
 import org.bukkit.OfflinePlayer;
+import org.bukkit.Server;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandExecutor;
 import org.bukkit.command.CommandSender;
-import org.bukkit.entity.Player;
+import org.popcraft.popcraft.newCode.PopCommand;
 import org.popcraft.popcraft.utils.Message;
+import org.slf4j.event.Level;
 
 import java.io.BufferedWriter;
 import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
 
-@Deprecated
+@PopCommand("listgen")
 public class Listgen implements CommandExecutor {
 
+    private final Server server;
+
+    @Inject
+    public Listgen(Server server) {
+        this.server = server;
+    }
+
     @Override
     public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
-        if (cmd.getName().equalsIgnoreCase("listgen")) {
-            try {
-                generatePlayerList();
-                generatePlayerCount();
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
-            if (sender instanceof Player) {
-                Message.normal((Player) sender, "List generation complete.");
-            } else {
-                Message.generic("List generation complete.");
-            }
+        OfflinePlayer[] playerList = this.server.getOfflinePlayers();
+        try {
+            generatePlayerList(playerList);
+            generatePlayerCount(playerList);
+        } catch (IOException e) {
+            Message.error(sender, "IOException");
             return true;
         }
-        return false;
+        Message.normal(sender, "List generation complete.");
+        return true;
     }
 
-    public void generatePlayerList() throws IOException {
-        OfflinePlayer[] playerlist = Bukkit.getServer().getOfflinePlayers();
+    public void generatePlayerList(OfflinePlayer[] playerList) throws IOException {
         File players = new File("players.txt");
-        if (!players.exists()) {
+        if (!players.exists())
             players.createNewFile();
-        }
-        FileWriter fw = new FileWriter(players.getAbsoluteFile());
-        BufferedWriter bw = new BufferedWriter(fw);
-        int playerlistlength = playerlist.length;
-        int n = 0;
-        while (n < playerlistlength) {
-            OfflinePlayer currentplayer = playerlist[n];
-            String username = currentplayer.getName();
-            bw.write(username);
+        BufferedWriter bw = new BufferedWriter(new FileWriter(players.getAbsoluteFile()));

```

Comment:
Wrap AutoCloseable resource within a Try-With-Resource block. For example:
```
try (final BufferedWriter bw = new BufferedWriter(new FileWriter(players.getAbsoluteFile()))) {
   //Other Stuff Goes Here
} catch (final IOException e) {
   plugin.log(Level.WARN, "Failed to Write out player list file", e);
}
```
Also consider checking out try (final BufferedWriter bw = Files.newBufferedReader(Paths.get("players.txt"))...